### PR TITLE
feature/iframe to master

### PR DESCRIPTION
**Pull request by NiclasNorin**

### CODE REVIEW

**Review by annalinneajohansson — COMMENTED** [review]
Kvarglömda console.log på rad 50 och 53 :)

**Comment by helsingborgsstad** [comment]
Task linked: [CU-3f2z3ph Hantera godkännande av iframe med javascript ](https://app.clickup.com/t/3f2z3ph)

**Review by sebastianthulin — CHANGES_REQUESTED** [review]

File: source/js/iframeAcceptance.js [hunk 1]
```diff
@@ -39,18 +39,27 @@ const onClicklHandler = (iframe) => {
...
-                    infoText: `This part of the website shows content from another website. By continuing, you are accepting GDPR and privacy policy.`,
-                    buttonText: "I understand, continue."
-                },
-                ...dataAttribute
+
+            const dataAttribute = JSON.parse(iframe.getAttribute('options') ?? '{}');
+            
+            const option = () => {
+                console.log("körs");
+                if (iframe.getAttribute('data-supplier') && iframe.getAttribute('data-policy')) {
```
[the remark this answers is not in the document]
Comment (NiclasNorin) [reply source/js/iframeAcceptance.js]: Func removed in latest commits

File: source/js/iframeAcceptance.js [hunk 2]
```diff
@@ -39,18 +39,27 @@ const onClicklHandler = (iframe) => {
...
+                   let options = dataAttribute.knownSupplierLabels;
+                   console.log(options);
+                    const supplier = options.infoText.replace('{SUPPLIER_WEBSITE}', iframe.getAttribute('data-supplier')).replace('{SUPPLIER_POLICY}', iframe.getAttribute('data-policy'));
```
[the remark this answers is not in the document]
Comment (NiclasNorin) [reply source/js/iframeAcceptance.js]: Removed and uses backend instead

File: source/js/iframeAcceptance.js [hunk 3]
```diff
@@ -39,18 +39,27 @@ const onClicklHandler = (iframe) => {
...
+                    options.infoText = supplier;
+                    return options;
+                } else {
+                    return dataAttribute.unknownSupplierLabels;
+                }
             }
-            const { labels: {titleText, infoText, buttonText }} = options; 
+
+            const {titleText, infoText, buttonText } = option(); 
             const div = document.createElement('div');
```
[the remark this answers is not in the document]
Comment (NiclasNorin) [reply source/js/iframeAcceptance.js:61]: Moved to backend

File: source/js/iframeAcceptance.js [hunk 4]
```diff
@@ -39,18 +39,27 @@ const onClicklHandler = (iframe) => {
...
             div.insertAdjacentHTML('beforeend', template({ titleText, infoText, buttonText }));
```
[the remark this answers is not in the document]
Comment (NiclasNorin) [reply source/js/iframeAcceptance.js]: Removed and moved to backend

File: source/js/iframeAcceptance.js [hunk 5]
```diff
@@ -1,14 +1,14 @@
 let acceptedSuppliers = JSON.parse(localStorage.getItem('acceptedSuppliers')) ?? [];
 
-const template = ({ titleText, infoText, buttonText }) => (`
-    <div class="js-suppressed-iframe-wrapper" style="position:relative;">
-        <div class="js-suppressed-iframe-prompt" style="position:absolute; left:0; top:0; width:100%; height:100%; z-index:1; background-color: white; display: flex; align-items: center;">
+const template = ({ title, info, button }) => (`
```
[the remark this answers is not in the document]
Comment (NiclasNorin) [reply source/js/iframeAcceptance.js:3]: Gets rendered in PHP now

**Pull request merged**